### ISSUE TITLE
Default untyped color

### ISSUE THREAD

**Issue by benloh** [netcreateorg/netcreate-itest#370]
@jdanish @kalanicraig for projects that do not define a node or edge type, should we use black/gray as the default color for the nodes and edges (current functionality)?  Or do you want to pick a default color, e.g. blue?

**Comment by jdanish** [comment]
A medium gray works well. The black stands out too much. Thanks!

**Comment by benloh** [comment]
# TO DO
- [ ] Change graph node default color to light gray
- [ ] Decide if untyped designation should be `undefined` or `--` (as selected by popup menu) or `null`?  Do we need to distinguish between an `un-reviewed`/`un-confirmed` state with an explicitly confirmed (decided-upon) `untyped` state?

**Comment by benloh** [comment]
@jdanish One additional question about default color/type behavior.  In the current system, when adding a new node or edge, type would be automatically set to the default blank value, which would result in the gray color.  The advantage of this is you could quickly define nodes and edges ignoring other values and rely on the node and edge defaulting to the "blank" default type.

But there is a bug associated with it: #411.  If you add a template with no default color, the system will act like the first color is selected, but it actually isn't.  So we in order to fix this, we need to explicitly set an "unselected" value.  We want to force users to select a type, e.g.:

<img width="284" height="158" alt="Image" src="https://github.com/user-attachments/assets/ffa679d0-9d19-4f17-8313-eedb2fd4dc40" />

...this way users are always explicitly asked to select a type.  And there is no such thing as a "blank" type value.  It will always point to a named type.  You can still plow through object creation, ignoring Type, but then you would see a "-- Select a Type --" message in the menu.  The other thing we can do is use the nodeDef.help field for the message.

If you really don't want a "Type", the solution would be to remove or hide the "Type" field altogether.

Any thoughts on this?  Or do you prefer the current behavior? 

If we wanted to do this...
- [ ] Review active templates:  Remove blank default types.
- [ ] Review default template:  Remove blank default types.
- [ ] Review templates: If you don't want to use `type`, remove it altogether (make sure that is gracefully handled)
- [ ] Review templates: If you don't want to use `type`, make sure merely hiding will work

**Comment by jdanish** [comment]
My vote:


   1. leave it as-is for now since if the default template has unselected
   as the top value, things work 99% of the time
   2. One of the issues for a future feature is to have optional /
   mandatory flags on attributes. Once we get that, it could be selectively
   used for this in educational contexts where we want to force it, and not in
   DH contexts where expects are likely doing what you suggested.

So ... I think we are good?

**Comment by benloh** [comment]
This is mainly a problem if you remove the blank `label=""` type.  Because of the way the menu is constructed (it reads the values defined in the type options, but does not create a blank "-- Select a category --" option) this means the first item gets auto-selected.  So if the first item is "People", it looks like it get selected, but the color will never be changed because we're not actually setting the value.  Worse, you can't select "People" unless you first select another type (e.g. "Place"), and the reselect "People".  

While this is only a problem if you remove the default type, this puts the onus on correct type option creation on the user/teacher, which doesn't seem ideal.

If we really want to the Node/Edge editor to auto-select the first item in the menu on creation, then we should explicitly set the value when loading, rather than relying on the menu option to select the first item.  And then we can get rid of the blank default if that's what we want to do.

So either way we want to do some modification of behaviors to better handle this.

**Comment by jdanish** [comment]
OK, then how about we:

a) add the code to auto-select the first item
b) leave the default and prior templates with the blank as previously setup.

In most cases, hopefully no one deletes that blank as it makes the most sense. But if they do, AOK.

For now, let's assume we want a type and it should be visible.

**Comment by benloh** [comment]
Ignore other TODOs!

# TO DO
- [x] If there is a "default" aka `""`` type defined, go ahead and select it.
- [x] If there isn't a "default" aka `""` type defined, inject a blank "" type.
   - [x] For colors, blank color is selected
   - [x] For text, a blank item is selected
- [x] TESTING
   - [x] Template with no default color type
   - [x] Template with no default generic type
   - [x] Template WITH default color type = ""
   - [x] Template WITH default generic type =""
